File: sdk/julia/src/partner.jl
# Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates
struct PartnerResource
    http::HttpClient
end

# GET /api/partner - Get partner info
function get_partner_info(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner"; params = params)
end

# POST /api/partner/apply - Apply to partner program
function apply(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/partner/apply"; params = params)
end

# GET /api/partner/tiers - List partner tiers
function list_tiers(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/tiers"; params = params)
end

# GET /api/partner/pricing - Get partner pricing
function get_pricing(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/pricing"; params = params)
end

# GET /api/partner/customers - List customers
function list_customers(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/customers"; params = params)
end

# GET /api/partner/earnings - Get earnings
function get_earnings(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/earnings"; params = params)
end

# GET /api/partner/wallet - Get wallet balance
function get_wallet(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/wallet"; params = params)
end

# GET /api/partner/payouts - List payouts
function list_payouts(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/payouts"; params = params)
end

# POST /api/partner/payouts - Request payout
function request_payout(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/partner/payouts"; params = params)
end

# GET /api/partner/leads - List leads
function list_leads(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/leads"; params = params)
end

# POST /api/partner/leads - Create lead
function create_lead(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/partner/leads"; params = params)
end

# GET /api/partner/referral - Get referral info
function get_referral(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/referral"; params = params)
end

# GET /api/partner/rates - Get exchange rates
function get_rates(r::PartnerResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/partner/rates"; params = params)
end

File: sdk/rust/src/partner.rs
//! Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct PartnerResource {
    http: Arc<HttpClient>,
}

impl PartnerResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/partner - Get partner info
    pub async fn get_partner_info(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner", None).await
    }

    /// POST /api/partner/apply - Apply to partner program
    pub async fn apply(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/partner/apply", None).await
    }

    /// GET /api/partner/tiers - List partner tiers
    pub async fn list_tiers(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/tiers", None).await
    }

    /// GET /api/partner/pricing - Get partner pricing
    pub async fn get_pricing(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/pricing", None).await
    }

    /// GET /api/partner/customers - List customers
    pub async fn list_customers(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/customers", None).await
    }

    /// GET /api/partner/earnings - Get earnings
    pub async fn get_earnings(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/earnings", None).await
    }

    /// GET /api/partner/wallet - Get wallet balance
    pub async fn get_wallet(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/wallet", None).await
    }

    /// GET /api/partner/payouts - List payouts
    pub async fn list_payouts(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/payouts", None).await
    }

    /// POST /api/partner/payouts - Request payout
    pub async fn request_payout(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/partner/payouts", None).await
    }

    /// GET /api/partner/leads - List leads
    pub async fn list_leads(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/leads", None).await
    }

    /// POST /api/partner/leads - Create lead
    pub async fn create_lead(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/partner/leads", None).await
    }

    /// GET /api/partner/referral - Get referral info
    pub async fn get_referral(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/referral", None).await
    }

    /// GET /api/partner/rates - Get exchange rates
    pub async fn get_rates(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/partner/rates", None).await
    }

}
